sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -610,10 +611,7 @@ export class EventHubSender extends LinkEntity {
           this._sender!.on(SenderEvents.rejected, onRejected);
           this._sender!.on(SenderEvents.modified, onModified);
           this._sender!.on(SenderEvents.released, onReleased);
-          waitTimer = setTimeout(
-            actionAfterTimeout,
-            Constants.defaultOperationTimeoutInSeconds * 1000
-          );
+          waitTimer = setTimeout(actionAfterTimeout, getRetryAttemptTimeoutInMs(options!.retryOptions));
```

Comment:
Just curious, was the TypeScript compiler unable to tell that options is always defined?

Reply:
Yeah, in the IDE I was seeing a squiggly prompt that it may be undefined..

Reply:
@chradek Would you prefer `options && options.retryOptions` over `options!.retryOptions`.
I am aware of the knee jerk reaction we get when we see the `!` :)
Was just curious about what you felt about the above alternative

Reply:
I actually wonder if changing
```
private _trySendBatch(
    message: AmqpMessage | Buffer,
    tag: any,
    options?: SendOptions & EventHubProducerOptions,
    format?: number
  ): Promise<void> {
    if (!options) {
      options = {};
    }
``` 
to

```
private _trySendBatch(
    message: AmqpMessage | Buffer,
    tag: any,
    options: SendOptions & EventHubProducerOptions = {},
    format?: number
  ): Promise<void> {
```
would solve the squiggles instead since then the compiler should see `options` always exists.

Reply:
Updating the type had worked